services/src/pro/datasets/postgres.rs

```diff
@@ -646,6 +647,32 @@ where
         Ok(DatasetIdAndName { id, name })
     }
 
+    async fn update_dataset_symbology(
+        &self,
+        dataset: DatasetId,
+        symbology: &Symbology,
+    ) -> Result<()> {
+        let mut conn = self.conn_pool.get().await?;
+
+        let tx = conn.build_transaction().start().await?;
+
+        ensure!(
+            self.has_permission_in_tx(dataset, Permission::Owner, &tx)
+                .await?,
+            error::PermissionDenied
+        );
+
+        let stmt = tx
+            .prepare("UPDATE datasets SET symbology = $2 WHERE id = $1;")
+            .await?;
+
+        tx.execute(&stmt, &[&dataset, &symbology]).await?;
```

Comment:
I guess you don't need to prepare if you only use this statement once.
From what I've read in the PostgreSQL package, this is safe because they use the query and the params using the protocol and not concatenating them into a string.